Adaptation of HTAN GeoMX Model for MC2/Spatial Transcriptomics Explorer

Bankso opened the pull request:
While this model is being introduced as part of work for the STE project, the intention is to provide metadata that supports future reuse and integration with other datasets. With this goal in mind, the current implementation retains all the features defined by the HTAN GeoMX model. Expectation is that this will be refined in collaboration with contributors.

Adds models:
Biospecimen
GeoMX Level 1 (FASTQs)
GeoMX Level 2 (DCC/RCC files)
GeoMX Imaging Level 2 (TIFF/OME-TIFF files)
GeoMX Level 3 (Counts matrices)
GeoMX ROI/Segment Annotation
Imaging Channel Metadata

Biospecimen model is based on features noted in Spatial Transcriptomics Explorer project proposal. Noted features were mapped to existing entries in the HTAN data model, but the attribute names and controlled vocabulary may require modification to fit expectations/data available from collaborators.

The NanoString GeoMX DSP data model is based on the HTAN model, with additional attributes and a FASTQ data level for the Spatial Transcriptomics Explorer project.

Bankso committed:
Add first draft of GeoMX model

Minimal first draft of the Nanostring GeoMX DSP metadata model for the Spatial Transcriptomics Explorer Portal.

Attributes were originally copied from the in-progress HTAN models for GeoMX datasets

Models defined:
- GeoMXLevel1
- ImagingLevel2
- GeoMXLevel3
- GeoMXROISegmentAnnotation

Bankso committed:
Add GeoMX level 0 model

Add components for FASTQ files (level 0) and rebuild CSV/JSONLD

Bankso committed:
Addition of files for GeoMX data model

Biospecimen model is based on features noted in Spatial Transcriptomics Explorer project proposal. Noted features were mapped to existing entries in the HTAN data model, but may require modification to fit data available from collaborators.

The NanoString GeoMX DSP data model is based on the HTAN model, with additional attributes and a FASTQ data level for the Spatial Transcriptomics Explorer project.

Adds models:
Biospecimen
GeoMX Level 1 (FASTQs)
GeoMX Level 2 (DCC/RCC files)
GeoMX Imaging Level 2 (TIFF/OME-TIFF files)
GeoMX Level 3 (Counts matrices)
GeoMX ROI/Segment Annotation
Imaging Channel Metadata

While this model is being introduced as part of work for the STE project, the intention is to provide metadata that supports future reuse and integration with other datasets. With this goal in mind, this initial commit retains all the features defined by the HTAN GeoMX model. Expectation is that this will be refined in collaboration with contributors/subject-matter experts.

Bankso committed:
Add CSV and JSON-LD

Generated locally due to issue with GitHub workflow

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Age at Diagnosis,Age at the time of diagnosis expressed in number of days since birth.,,,FALSE,,,,HTAN,
+Biospecimen Tumor Type,"The type of tumor associated with the specimen. Multiple values permitted, comma separated.",,,FALSE,,,,MC2,
+Tumor Subtype,"The subtype related to the scientific determination and investigation, analysis and recognition of the presence and nature of disease, condition, or injury from expressed signs and symptoms of tissue growth resulting from uncontrolled cell proliferation.",,,FALSE,,,,,
+Tumor Grade,"Numeric value to express the degree of abnormality of cancer cells, a measure of differentiation and aggressiveness.",,,FALSE,,,,HTAN,
```
Added "Tumor Grade" to address Tumor Stage/Disease Progression feature

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Biospecimen Tumor Type,"The type of tumor associated with the specimen. Multiple values permitted, comma separated.",,,FALSE,,,,MC2,
+Tumor Subtype,"The subtype related to the scientific determination and investigation, analysis and recognition of the presence and nature of disease, condition, or injury from expressed signs and symptoms of tissue growth resulting from uncontrolled cell proliferation.",,,FALSE,,,,,
+Tumor Grade,"Numeric value to express the degree of abnormality of cancer cells, a measure of differentiation and aggressiveness.",,,FALSE,,,,HTAN,
+Metastasis at Diagnosis,The text term used to describe the extent of metastatic disease present at diagnosis.,,,FALSE,,,,HTAN,
```
Added "Metastasis at Diagnosis" to address Tumor Stage/Disease Progression feature

jaybee84 replied:
Is this expected to be a boolean?

Bankso replied:
Nope, this attribute would be populated using valid values. These are the terms used for this attribute, per the HTAN model: https://github.com/mc2-center/data-models/pull/22/files#diff-950e2286f9cf7ee7a9f9eb7de867a511d2483bedcf1bf750895df11bbc737c69

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Tumor Subtype,"The subtype related to the scientific determination and investigation, analysis and recognition of the presence and nature of disease, condition, or injury from expressed signs and symptoms of tissue growth resulting from uncontrolled cell proliferation.",,,FALSE,,,,,
+Tumor Grade,"Numeric value to express the degree of abnormality of cancer cells, a measure of differentiation and aggressiveness.",,,FALSE,,,,HTAN,
+Metastasis at Diagnosis,The text term used to describe the extent of metastatic disease present at diagnosis.,,,FALSE,,,,HTAN,
+Last Known Disease Status,Text term that describes the last known state or condition of an individual's neoplasm.,,,FALSE,,,,HTAN,
```
Added "Last Known Disease Status" to address Tumor Stage/Disease Progression feature

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Metastasis at Diagnosis,The text term used to describe the extent of metastatic disease present at diagnosis.,,,FALSE,,,,HTAN,
+Last Known Disease Status,Text term that describes the last known state or condition of an individual's neoplasm.,,,FALSE,,,,HTAN,
+Preservation Method,Text term that represents the method used to preserve the sample.,,,FALSE,,,,HTAN,
+Site of Resection or Biopsy,"The text term used to describe the anatomic site of the resection or biopsy of the patient's malignant disease, as described by the World Health Organization's (WHO) International Classification of Diseases for Oncology (ICD-O).",,,FALSE,,,,HTAN,
```
Added "Site of Resection or Biopsy" to address Tumor Location feature

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Last Known Disease Status,Text term that describes the last known state or condition of an individual's neoplasm.,,,FALSE,,,,HTAN,
+Preservation Method,Text term that represents the method used to preserve the sample.,,,FALSE,,,,HTAN,
+Site of Resection or Biopsy,"The text term used to describe the anatomic site of the resection or biopsy of the patient's malignant disease, as described by the World Health Organization's (WHO) International Classification of Diseases for Oncology (ICD-O).",,,FALSE,,,,HTAN,
+Tissue or Organ of Origin,"The text term used to describe the anatomic site of origin, of the patient's malignant disease, as described by the World Health Organization's (WHO) International Classification of Diseases for Oncology (ICD-O).",,,FALSE,,,,HTAN,
```
Added "Tissue or Organ of Origin" to address Tumor Location feature

jaybee84 replied:
This term is a bit ambiguous. Tissue expects values like "epithelial/ mesodermal/ stromal" while Organ expects values like "kidney/ brain/ breast" . Which is the target here? Suggest making the term more specific.

Bankso replied:
I agree, there is some ambiguity here, though this is how the term is listed and defined within the HTAN model. "Tumor Origin Site" may be a better fit, given the set of valid values available in the MC2 data model, which I expect we will be using for "Site of Resection or Biopsy" as well: https://github.com/mc2-center/data-models/blob/main/modules/shared/tissue.csv

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+GeoMx DSP Library Preparation Kit Version,Version of Library Preparation Kit. String,,,FALSE,,,,HTAN,
+NanoString GeoMx DSP Spatial Transcriptomics Level 2,Raw counts (DCC/RCC) files from GeoMX experiments,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of Parent FASTQ File, Synapse ID of GeoMx Lab Worksheet File, Synapse ID of GeoMx DSP PKC File, Synapse ID of GeoMx DSP ROI Segment Annotation File, GeoMx DSP Assay Type, ROI Name, AOI Name, Software a ...
+Synapse ID of Parent FASTQ File,Synapse ID(s) indicating the FASTQ file(s) from which the DCC file was derived; multiple parent files should be comma-separated,,,FALSE,,,,HTAN,
+NanoString GeoMx DSP Imaging Level 2,,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of GeoMx DCC File, Synapse ID of GeoMx ROI Segment Annotation File, ROI Name, AOI Name, Channel Metadata Filename, Imaging Assay Type, Protocol Link, Software and Version, Microscope, Objective, NominalMagnification, LensNA, WorkingDistance, WorkingDistanceUnit, Imm ...
```
We had discussed removing some of these attributes previously, but maintaining them all seems like a good option if we want to ensure the GeoMX image files are compatible with data from other experiments.

Bankso replied:
I added "Synapse ID of GeoMx DCC File" because I think it makes sense to reference DCC file(s) associated with the image, but I'm not completely sure if this is necessary.

"ROI Name" and "AOI Name" were added to help make it clear which regions in the ROI annotation file are associated with the image.

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+GeoMx DSP Unique Probe Count,Total number of unique probes reported.,,,FALSE,,,,HTAN,
+GeoMx DSP Unique Target Count,Total number of unique genes reported.,,,FALSE,,,,HTAN,
+Matrix Type,Type of data stored in matrix.,,,FALSE,,,,HTAN,
+GeoMX ROI Segment Annotation,"GeoMx ROI and Segment Metadata Attributes. The assayed biospecimen should be reported one per row with the associated ROI and AOI coordinates. ",,"Component, Parent Biospecimen ID, Scan name, ROI Name, AOI Name, Tags, ROI X Coordinate, ROI Y Coordinate, AOI X Coordinate, AOI Y Coordinate, QC status, Scan Height, Scan Width, Scan Offset X, Scan Offset Y, Binding Densi ...
```
This is a combination of features from the RCC and DCC versions of the ROI annotation file defined in the HTAN model. We can define dependent attributes to indicate which ones are required for each file format, but I wanted to make sure we have the features available.

Since there can be multiple AOIs within an ROI, I added coordinates for AOI regions as well. This can be removed if unnecessary.

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+SizeY,Size of image: Y dimension (in pixels). Integer >= 1,,,FALSE,,,,HTAN,
+SizeZ,Size of image: Z dimension (in pixels). Integer >= 1,,,FALSE,,,,HTAN,
+PixelType,"Data type for each pixel value. E.g. ""uint16""",,,FALSE,,,,HTAN,
+NanoString GeoMx DSP Spatial Transcriptomics Level 3,Files contain processed data from the NanoString GeoMx DSP Pipeline.,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of Parent Data File, Synapse ID GeoMx ROI Segment Annotation File, GeoMx DSP Unique Probe Count, GeoMx DSP Unique Target Count, Matrix Type, GeoMx DSP Genomic Reference, GeoMx DSP Wo ...
```
This should be the same as the model defined by HTAN.

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Biological probe median,Is the median count from all probes except the negative control probes. A measure of signal to background for each segment,,,FALSE,,,,HTAN,
+Sequencing Saturation,"The fraction of reads originating from an already-observed UMI. This is a function of library complexity and sequencing depth. More specifically, this is the fraction of confidently mapped, valid spot-barcode, valid UMI reads that had a non-unique (spot-barcode, UMI, gene).",,,FALSE,,,,HTAN,
+MapQ30,Number of reads with Quality >= 30.,,,FALSE,,,,HTAN,
+Imaging Channel Metadata,Channel-level Metadata Attributes,,"Component, Channel Metadata ID, Channel ID, Channel Name, Channel Passed QC, Cycle Number, Sub Cycle Number, Antibody Role, Target Name, Antibody Name, RRID identifier, Fluorophore, Clone, Lot, Vendor, Catalog Number, Excitation Wavelength, Emission Wavelength, Excitation Bandwidth, Emission Bandwidth, Metal Isotope Element, Metal Isoto ...
```
I was going back and forth on including this, but as I mentioned above, I imagine that this information would be helpful for future reuse of the imaging data.

jaybee84 replied:
Agree. But I wonder if these metadata come as a separate csv file like the `config.ini` file. In that case capturing the metadata file should suffice and we wont need these columns for each file.

Bankso replied:
"Imaging Channel Metadata" is defined on line 154 as a separate metadata file, so the information can be collected and then referenced by other manifests.

adamjtaylor replied:
Are you expecting to only have H&E images? If so channel metadata can be dropped. If you are expecting IF data lets think of a better way as the channel metadata file approach used by HTAN is very limited.

Bankso replied:
Thanks for the insight! We are definitely expecting IF, so I'll look into other approaches for capturing this info

adamjtaylor replied:
Propose replacing `Imaging Channel Metadata` with `Imaging Channel Names` which is a list like entry of the channel names / targets used. For stretch could validate these with valid values.

Bankso commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Tissue or Organ of Origin,"The text term used to describe the anatomic site of origin, of the patient's malignant disease, as described by the World Health Organization's (WHO) International Classification of Diseases for Oncology (ICD-O).",,,FALSE,,,,HTAN,
+Timepoint Label,"Label to identify the time point at which the clinical data or biospecimen was obtained (e.g. Baseline, End of Treatment, Overall survival, Final). NO PHI/PII INFORMATION IS ALLOWED.",,,FALSE,,,,HTAN,
+Health Care Site,The health care site at which the biospecimen sample was collected.,,,FALSE,,,,,
+NanoString GeoMx DSP Spatial Transcriptomics Level 1,Raw sequencing files from GeoMX experiments,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of GeoMx Lab Worksheet File, Synapse ID of GeoMx DSP PKC File, Synapse ID of GeoMx DSP ROI Segment Annotation File, GeoMx DSP Assay Type, Target Depth, Read Length, ROI Name, AOI Name, Software and Version,  ...
```
The FASTQ (level 1) attributes are mostly the same as requested for the DCC file (level 2), except "Target Depth" and "Read Length".

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Tumor Subtype,"The subtype related to the scientific determination and investigation, analysis and recognition of the presence and nature of disease, condition, or injury from expressed signs and symptoms of tissue growth resulting from uncontrolled cell proliferation.",,,FALSE,,,,,
+Tumor Grade,"Numeric value to express the degree of abnormality of cancer cells, a measure of differentiation and aggressiveness.",,,FALSE,,,,HTAN,
+Metastasis at Diagnosis,The text term used to describe the extent of metastatic disease present at diagnosis.,,,FALSE,,,,HTAN,
+Last Known Disease Status,Text term that describes the last known state or condition of an individual's neoplasm.,,,FALSE,,,,HTAN,
```
There seems to be some redundancy between this term and the previous "Metastatic at diagnosis". Suggest keeping this term instead of the previous one, unless there is a compelling use case.

Bankso replied:
I agree, there is some overlap between "Metastasis at Diagnosis" and "Last Known Disease Status". I saw "Last Known Disease Status" as centered on disease progression/outcome, independent of initial metastasis status. Here are the valid values available in the HTAN model: https://github.com/mc2-center/data-models/pull/22/files#diff-a5419206084b2622b5231f292c8a2103bb8f4a5b97bf51ef95e3070aec43dada

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Preservation Method,Text term that represents the method used to preserve the sample.,,,FALSE,,,,HTAN,
+Site of Resection or Biopsy,"The text term used to describe the anatomic site of the resection or biopsy of the patient's malignant disease, as described by the World Health Organization's (WHO) International Classification of Diseases for Oncology (ICD-O).",,,FALSE,,,,HTAN,
+Tissue or Organ of Origin,"The text term used to describe the anatomic site of origin, of the patient's malignant disease, as described by the World Health Organization's (WHO) International Classification of Diseases for Oncology (ICD-O).",,,FALSE,,,,HTAN,
+Timepoint Label,"Label to identify the time point at which the clinical data or biospecimen was obtained (e.g. Baseline, End of Treatment, Overall survival, Final). NO PHI/PII INFORMATION IS ALLOWED.",,,FALSE,,,,HTAN,
```
```suggestion
Timepoint,"Label to identify the time point at which the clinical data or biospecimen was obtained (e.g. Baseline, End of Treatment, Overall survival, Final). NO PHI/PII INFORMATION IS ALLOWED.",,,FALSE,,,,HTAN,
```

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+File Format,"Format of a file (e.g. txt, csv, fastq, bam, etc.)",,,TRUE,,,,HTAN,
+Parent Biospecimen ID,Biospecimen Identifier indicating the biospecimen(s) from which these files were derived; multiple parent biospecimen should be comma-separated,,,TRUE,,,,HTAN,
+Synapse ID of Data File,Synapse ID of file,,,TRUE,,,,HTAN,
+Synapse ID of GeoMx Lab Worksheet File,Synapse ID(s) of Lab Worksheet Files output from the GeoMx DSP workflow. Multiple files are listed as comma separated values.,,,FALSE,,,,HTAN,
```
These files seem to suggest linking of resources using provenance : 
Synapse ID of GeoMx Lab Worksheet File
Synapse ID of GeoMx DSP PKC File
Synapse ID of GeoMx DSP ROI Segment Annotation File

These should be added using synapseclient's "addActivity " feature instead of annotation. 

Is the purpose of including them here to allow collecting the mapping of these activities?

Bankso replied:
Yes, my intention was to include attributes that point to the relevant accessory files that we expect to be uploaded.

When you say: 

> These should be added using synapseclient's "addActivity " feature instead of annotation.

Are you referring to the STE portal data or in general? I'm not sure that this is the route we plan to go for MC2 - our path forward with respect to resource linking is still being worked out

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+GeoMx DSP Assay Type,The assay type which was used for the GeoMx DSP pipeline.,,,FALSE,,,,HTAN,
+Target Depth,The targeted read depth prior to sequencing. Integer,,,TRUE,,,,HTAN,
+Read Length,"The length of the sequencing reads. Can be integer, null",,,TRUE,,,,HTAN,
+ROI Name,Name of Region of Interest corresponding to file. Multiple names should be comma-separated,,,TRUE,,,,HTAN,
```
Suggest not allowing multiple values here, since the ROIs should be fixed for each file.

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+GeoMx DSP NGS Library Preparation Kit Name,Name of Library Preparation Kit. String,,,FALSE,,,,HTAN,
+GeoMx DSP Library Preparation Kit Vendor,Vendor of Library Preparation Kit. String,,,FALSE,,,,HTAN,
+GeoMx DSP Library Preparation Kit Version,Version of Library Preparation Kit. String,,,FALSE,,,,HTAN,
+NanoString GeoMx DSP Spatial Transcriptomics Level 2,Raw counts (DCC/RCC) files from GeoMX experiments,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of Parent FASTQ File, Synapse ID of GeoMx Lab Worksheet File, Synapse ID of GeoMx DSP PKC File, Synapse ID of GeoMx DSP ROI Segment Annotation File, GeoMx DSP Assay Type, ROI Name, AOI Name, Software a ...
```
The DCC files are level 2 files which are similar to "quant.sf" files for regular RNAseq data. 

A "summary.txt" file is expected which aggregates all DCC files. This should be collected in lieu of individual DCC files if possible to reduce the burden of data contributors. This will be multispecimen, so multiple specimen IDs should be expected as metadata for the file.

jaybee84 replied:
Similarly, a config.ini file should be expected to be uploaded. 

From a re-use perspective, the config.ini file + fastq.gz files together can recreate the DCC files if needed.

Bankso replied:
> A "summary.txt" file is expected which aggregates all DCC files.

This seems reasonable to collect, I agree. I think it will be good to have a manifest for individual DCC/RCC file metadata and a manifest for the "summary.txt" available, to accommodate either approach.

>Similarly, a config.ini file should be expected to be uploaded.

This makes sense for the fastq.gz/DCC route, but I don't think this applies to RCC files, so we will need to account for this in the model.

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+GeoMx DSP Library Preparation Kit Vendor,Vendor of Library Preparation Kit. String,,,FALSE,,,,HTAN,
+GeoMx DSP Library Preparation Kit Version,Version of Library Preparation Kit. String,,,FALSE,,,,HTAN,
+NanoString GeoMx DSP Spatial Transcriptomics Level 2,Raw counts (DCC/RCC) files from GeoMX experiments,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of Parent FASTQ File, Synapse ID of GeoMx Lab Worksheet File, Synapse ID of GeoMx DSP PKC File, Synapse ID of GeoMx DSP ROI Segment Annotation File, GeoMx DSP Assay Type, ROI Name, AOI Name, Software a ...
+Synapse ID of Parent FASTQ File,Synapse ID(s) indicating the FASTQ file(s) from which the DCC file was derived; multiple parent files should be comma-separated,,,FALSE,,,,HTAN,
```
This seems to be a provenance item.

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+NanoString GeoMx DSP Spatial Transcriptomics Level 2,Raw counts (DCC/RCC) files from GeoMX experiments,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of Parent FASTQ File, Synapse ID of GeoMx Lab Worksheet File, Synapse ID of GeoMx DSP PKC File, Synapse ID of GeoMx DSP ROI Segment Annotation File, GeoMx DSP Assay Type, ROI Name, AOI Name, Software a ...
+Synapse ID of Parent FASTQ File,Synapse ID(s) indicating the FASTQ file(s) from which the DCC file was derived; multiple parent files should be comma-separated,,,FALSE,,,,HTAN,
+NanoString GeoMx DSP Imaging Level 2,,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of GeoMx DCC File, Synapse ID of GeoMx ROI Segment Annotation File, ROI Name, AOI Name, Channel Metadata Filename, Imaging Assay Type, Protocol Link, Software and Version, Microscope, Objective, NominalMagnification, LensNA, WorkingDistance, WorkingDistanceUnit, Imm ...
+Synapse ID of GeoMx DCC File,Synapse ID(s) for related DCC file(s) derived from the GeoMx DSP experiment.,,,FALSE,,,,,
```
Recommend linking images to original Fastq files instead of DCC files.

Bankso replied:
I had anticipated linking images to DCC/RCC files, since fastq files won't be available if the RCC format is used. Though I do understand where you're coming from - this won't work if we are only collecting DCC "summary.txt" files. I wonder if it would make sense to link everything via ROI/AOI names, to avoid this issue?

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Synapse ID of Parent FASTQ File,Synapse ID(s) indicating the FASTQ file(s) from which the DCC file was derived; multiple parent files should be comma-separated,,,FALSE,,,,HTAN,
+NanoString GeoMx DSP Imaging Level 2,,,"Component, Filename, File Format, Parent Biospecimen ID, Synapse ID of Data File, Synapse ID of GeoMx DCC File, Synapse ID of GeoMx ROI Segment Annotation File, ROI Name, AOI Name, Channel Metadata Filename, Imaging Assay Type, Protocol Link, Software and Version, Microscope, Objective, NominalMagnification, LensNA, WorkingDistance, WorkingDistanceUnit, Imm ...
+Synapse ID of GeoMx DCC File,Synapse ID(s) for related DCC file(s) derived from the GeoMx DSP experiment.,,,FALSE,,,,,
+Channel Metadata Filename,Full path within Synapse project of uploaded companion CSV file containing channel-level metadata details,,,FALSE,,,,HTAN,
```
This one + config.ini files are expected metadata files that should be captured. 

I want to emphasize that these two should be considered metadata files and not data files, potentially requiring fewer metadata fields

jaybee84 commented on mc2.model.csv:
```diff
@@ -47,6 +47,136 @@ Resource Access Hazard,"Sensory hazards (Flashing, Motion, etc.) applicable to t
+Channel Metadata Filename,Full path within Synapse project of uploaded companion CSV file containing channel-level metadata details,,,FALSE,,,,HTAN,
+Imaging Assay Type,Type of imaging assay,,,FALSE,,,,HTAN,
+Protocol Link,"Protocols.io ID or DOI link to a free/open protocol resource describing in detail the assay protocol (e.g. surface markers used in Smart-seq, dissociation duration,  lot/batch numbers for key reagents such as primers, sequencing reagent kits, etc.) or the protocol by which the sample was obtained or generated.",,,FALSE,,,,HTAN,
+Microscope,"Microscope type (manufacturer, model, etc) used for this experiment",,,FALSE,,,,HTAN,
```
I would urge review of the microscope, lens, and magnification related fields for ST data. I imagine that the `platform`  will have fixed imaging configurations. So these fields may not be necessary to capture. The `platform` or the associated `imagingLabworksheet` will be the proxy for these metadata.

Bankso replied:
I agree, many of these fields will be fixed, I'm still trying to find a source for many of the details.

I did find this GEO listing, suggesting the microscope uses 20x magnification: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4844875

jaybee84 reviewed (changes requested):
@Bankso Thanks for getting this started. I added a few comments. Please let me know if you have any questions.

Bankso committed:
Initial drafts of new models related to STE work

- separated Biospecimen model from GeoMX
- separated Imaging level 2 from GeoMX
- reduced number of terms included in GeoMX imaging level 2
- updated valid values, generated model csv and json-ld

Bankso committed:
Add current manifest csvs

Bankso committed:
Remove outputs to prep for merge into main

Bankso committed:
Merge branch 'main' into ste-model

Bankso committed:
Rebuild model files
